Amplify/Categories/DataStore/Query/ModelKey.swift

```diff
@@ -32,24 +32,43 @@ import Foundation
 ///     post.content != nil
 /// })
 /// ```
-public protocol ModelKey: CodingKey, CaseIterable, QueryFieldOperation {}
+public protocol ModelKey: CodingKey, CaseIterable, QueryFieldOperation, DefaultLogger {
+    var modelName: String { get }
+}
+
+extension ModelKey {
+    public var modelName: String { "" }
+}
 
 extension CodingKey where Self: ModelKey {
 
+    var columnName: String {
+        guard let modelSchema: ModelSchema = ModelRegistry.modelSchema(from: modelName) else {
+            log.warn("Please upgrade to the latest version of Amplify CLI and rerun `amplify codegen models`")
+            return stringValue
+        }
+        switch modelSchema.field(withName: stringValue)?.association {
+        case .belongsTo(_, let targetName):
+            return targetName ?? stringValue
```

Comment:
We seem to have a unit tests which tests when targetName is not nil, but we should add a test where targetName is nil, so that it just returns the regular `stringValue`.